clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -544,25 +544,17 @@ static ControlFlowKind CheckFallThrough(AnalysisDeclContext &AC) {
 namespace {
 
 struct CheckFallThroughDiagnostics {
-  unsigned diag_MaybeFallThrough_HasNoReturn;
-  unsigned diag_MaybeFallThrough_ReturnsNonVoid;
-  unsigned diag_AlwaysFallThrough_HasNoReturn;
-  unsigned diag_AlwaysFallThrough_ReturnsNonVoid;
+  unsigned diag_FallThrough_HasNoReturn;
+  unsigned diag_FallThrough_ReturnsNonVoid;
   unsigned diag_NeverFallThroughOrReturn;
```

Comment:
While we’re at it, could you do a small cleanup pass here and just default these diag IDs to `0` (i.e. `= 0` here) and then only assign a value to the ones we care about in each of the ctors instead of having `= 0` there all over the place. That’d prevent us from accidentally forgetting to initialise one of these.